- Objective: find stores in Pittsburgh that don't offer SNAP

- Input files: 1) snap_stores.csv 2) pgh_stores.csv

- Process: convert all store names to lower case in both datasets. find pittsburgh stores that are not in SNAP stores. create a set of these stores to get unique store names. 

- Findings: there are 917 SNAP stores and 123 stores in pittsburgh. out of the 123, there are 116 that don't offer SNAP. the next step is to visualize the locations of these stores and see if they are in proximity to snap beneficiaries - if yes, then there's a strong case for making these stores offer SNAP benefits.

In [20]:
import pandas as pd

In [21]:
snap_stores = pd.read_csv('SNAP_Store_Locations.csv')
snap_stores.tail()

X          Y            Store_Name                Address  \
924 -79.908386  40.408356  DOLLAR GENERAL  9694          325 E 8th Ave   
925 -79.912804  40.405575       WALGREENS 09674          133 W 8th Ave   
926 -79.910011  40.412422      GIANT EAGLE 6379    420 Waterfront Dr E   
927 -79.909271  40.388477     Coen Markets 1218          3439 W Run Rd   
928 -79.901901  40.409134         RITE AID 4682  100 William Marks Way   

    Address_Line__2            City State   Zip5     County  Longitude  \
924             NaN       Homestead    PA  15120  ALLEGHENY -79.908386   
925             NaN       Homestead    PA  15120  ALLEGHENY -79.912804   
926             NaN       Homestead    PA  15120  ALLEGHENY -79.910011   
927             NaN  West Homestead    PA  15120  ALLEGHENY -79.909271   
928             NaN         Munhall    PA  15120  ALLEGHENY -79.901901   

      Latitude  ObjectId  
924  40.408356    244544  
925  40.405575    244545  
926  40.412422    244546  
927  40.388477    244547  
928  40.409134    244548

In [22]:
snap_stores.shape

(929, 12)

In [23]:
snap_stores.describe()

X           Y          Zip5   Longitude    Latitude  \
count  929.000000  929.000000    929.000000  929.000000  929.000000   
mean   -79.955007   40.440113  15169.551130  -79.955007   40.440113   
std      0.112884    0.076241     63.262536    0.112884    0.076241   
min    -80.351875   40.264488  15014.000000  -80.351875   40.264488   
25%    -80.032509   40.387852  15122.000000  -80.032509   40.387852   
50%    -79.958954   40.439213  15205.000000  -79.958954   40.439213   
75%    -79.876122   40.478672  15221.000000  -79.876122   40.478672   
max    -79.705376   40.659119  15275.000000  -79.705376   40.659119   

            ObjectId  
count     929.000000  
mean   128150.062433  
std     73956.116266  
min       350.000000  
25%     65354.000000  
50%    126322.000000  
75%    192277.000000  
max    244548.000000

In [24]:
print(snap_stores.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 929 entries, 0 to 928
Data columns (total 12 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   X                929 non-null    float64
 1   Y                929 non-null    float64
 2   Store_Name       929 non-null    object 
 3   Address          929 non-null    object 
 4   Address_Line__2  34 non-null     object 
 5   City             929 non-null    object 
 6   State            929 non-null    object 
 7   Zip5             929 non-null    int64  
 8   County           929 non-null    object 
 9   Longitude        929 non-null    float64
 10  Latitude         929 non-null    float64
 11  ObjectId         929 non-null    int64  
dtypes: float64(4), int64(2), object(6)
memory usage: 87.2+ KB
None


The first dataset on SNAP stores have 929 observations where each row is a unique Store location and 9 columns that include the latitude, longitude, store name, address, city, state, zip code, county, and unique identifier.

In [25]:
pgh_stores = pd.read_csv('PGH_supermarkets.csv')
pgh_stores.head()

name               street_address        city state  \
0          Kuhn's Market              661 Lincoln Ave  Pittsburgh    PA   
1            Food Shoppe  5878 Northumberland  Street  Pittsburgh    PA   
2  Shop 'n Save / Shaler      1620 Babcock  Boulevard  Pittsburgh    PA   
3        Giant Eagle #61         2021 Wharton  Street  Pittsburgh    PA   
4         Safrans Market            429 Walnut Street   Sewickley    PA   

   zip_code   latitude  longitude localizability  \
0     15202  40.497162 -80.060279          fixed   
1     15217  40.441561 -79.920536          fixed   
2     15209  40.499147 -79.978901          fixed   
3     15203  40.431132 -79.976607          fixed   
4     15143  40.541039 -80.182262          fixed   

                                    data_source_name  \
0  WPRDC Dataset: Allegheny County Restaurant/Foo...   
1  WPRDC Dataset: Allegheny County Restaurant/Foo...   
2  WPRDC Dataset: Allegheny County Restaurant/Foo...   
3  WPRDC Dataset: Allegheny County Restaurant/Foo...   
4  WPRDC Dataset: Allegheny County Restaurant/Foo...   

                                     data_source_url  primary_key_from_rocket  \
0  https://data.wprdc.org/dataset/allegheny-count...                    36519   
1  https://data.wprdc.org/dataset/allegheny-count...                    36498   
2  https://data.wprdc.org/dataset/allegheny-count...                    32725   
3  https://data.wprdc.org/dataset/allegheny-count...                    38879   
4  https://data.wprdc.org/dataset/allegheny-count...                    30265   

     asset_type  
0  supermarkets  
1  supermarkets  
2  supermarkets  
3  supermarkets  
4  supermarkets

In [26]:
pgh_stores.shape

(133, 12)

In [27]:
pgh_stores.columns

Index(['name', 'street_address', 'city', 'state', 'zip_code', 'latitude',
       'longitude', 'localizability', 'data_source_name', 'data_source_url',
       'primary_key_from_rocket', 'asset_type'],
      dtype='object')

In [28]:
pgh_stores.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 133 entries, 0 to 132
Data columns (total 12 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   name                     133 non-null    object 
 1   street_address           133 non-null    object 
 2   city                     133 non-null    object 
 3   state                    133 non-null    object 
 4   zip_code                 133 non-null    int64  
 5   latitude                 115 non-null    float64
 6   longitude                115 non-null    float64
 7   localizability           133 non-null    object 
 8   data_source_name         133 non-null    object 
 9   data_source_url          133 non-null    object 
 10  primary_key_from_rocket  133 non-null    int64  
 11  asset_type               133 non-null    object 
dtypes: float64(2), int64(2), object(8)
memory usage: 12.6+ KB


In [29]:
pgh_stores.describe()

zip_code    latitude   longitude  primary_key_from_rocket
count    133.000000  115.000000  115.000000             1.330000e+02
mean   15170.481203   40.446749  -79.964689             1.178357e+11
std       63.718557    0.082790    0.118007             9.932383e+10
min    15017.000000   40.279924  -80.241229             2.530000e+02
25%    15122.000000   40.382869  -80.045006             4.241200e+04
50%    15205.000000   40.441638  -79.964915             2.003082e+11
75%    15224.000000   40.493213  -79.874512             2.011022e+11
max    15275.000000   40.650507  -79.704765             2.019061e+11

The second dataset on grocery stores in Pittsburgh has 133 observations where each row is a unique store location and 12 columns including store name, address, city, state, zip code, latitude, longitude, localizability, data source, data source url, primary key, and asset type (type of store).

In [30]:
# snap_stores.drop(columns=['Match'],inplace=True)

In [31]:
# values = set(pgh_stores['name'].lower())

# snap_stores['Match'] = snap_stores['Store_Name'].lower().isin(values)
# snap_stores['Match'].value_counts()


#create new columns with all lower case store names in both datasets
snap_stores['store_snap'] = snap_stores['Store_Name'].str.lower()
pgh_stores['store_pgh'] = pgh_stores['name'].str.lower()

In [32]:
#find total unique stores in pittsburgh
pgh_stores_list = set(list(pgh_stores.store_pgh))
len(pgh_stores_list)

123

In [33]:
#snap_stores['store_snap'].value_counts()

#find out how many pittsburgh stores are in snap stores df
pgh_stores['store_pgh_snap'] = pgh_stores['store_pgh'].isin(set(snap_stores['store_snap']))
pgh_stores['store_pgh_snap'].value_counts() 

#there are 12 stores in pittsburgh that are in snap stores df - these don't add up to 123 because there are duplicates (one store - multiple locations)

False    121
True      12
Name: store_pgh_snap, dtype: int64

In [34]:
# #df of pittsburgh stores in snap stores
# pgh_stores[pgh_stores['store_pgh_snap'] == True]

In [35]:
#pgh_stores[pgh_stores['store_pgh_snap'] == 1]

#generate list of pgh stores that are NOT snap
pgh_stores_notsnap = pgh_stores[pgh_stores['store_pgh_snap'] == 0]
pgh_stores_notsnap.drop(pgh_stores.columns.difference(['store_pgh']), 1, inplace=True)

C:\Users\mahru\AppData\Local\Temp\ipykernel_21392\2550447277.py:5: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only.
  pgh_stores_notsnap.drop(pgh_stores.columns.difference(['store_pgh']), 1, inplace=True)
C:\Users\mahru\AppData\Local\Temp\ipykernel_21392\2550447277.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  pgh_stores_notsnap.drop(pgh_stores.columns.difference(['store_pgh']), 1, inplace=True)


In [36]:
#create a list of above, convert to set to get unique stores in pittsburgh that are not SNAP
pgh_stores_notsnap_list = set(list(pgh_stores_notsnap.store_pgh))
len(pgh_stores_notsnap_list)

116

In [37]:
#get a list of unique snap stores
snap_stores.drop(snap_stores.columns.difference(['store_snap']), 1, inplace=True)
snap_stores_list = set(list(snap_stores.store_snap))
len(snap_stores_list)

C:\Users\mahru\AppData\Local\Temp\ipykernel_21392\3370884206.py:2: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only.
  snap_stores.drop(snap_stores.columns.difference(['store_snap']), 1, inplace=True)


917